Angular/src/app/app.component.ts

```diff
@@ -1,13 +1,19 @@
 import { Component } from '@angular/core';
 import { RouterOutlet } from '@angular/router';
+import { LoaderComponent } from './components/loader/loader.component';
+import { LoadingService } from './services/loading/loading.service';
 
 @Component({
   selector: 'app-root',
   standalone: true,
-  imports: [RouterOutlet],
+  imports: [RouterOutlet, LoaderComponent],
   templateUrl: './app.component.html',
   styleUrl: './app.component.scss'
 })
 export class AppComponent {
   title = 'angular-gemini-recipes';
+
+  constructor(
+    public loadingService: LoadingService
```

Comment:
I usually try to keep my injected services `private` and `readonly`:

```suggestion
    private readonly loadingService: LoadingService
```

Reply:
https://ts.dev/style/#use-readonly

Reply:
i tried it, but getting:

`Property 'loadingService' is private and only accessible within class 'AppComponent'.ngtsc(2341)
app.component.ts(5, 56): Error occurs in the template of component AppComponent.
private (property) AppComponent.loadingService: LoadingService`

Reply:
You are still using it in the template. Don't use it there. The Loading component should inject the service and handle its visibility. No need to pass any inputs. Also calling functions from a template is not recommended.

Reply:
done

`constructor(
    public readonly loadingService: LoadingService
  ) {}`